packages/vscode-extension/src/service/ListFilesService.ts

```diff
@@ -0,0 +1,127 @@
+		this.executionPath = getRootProjectFolder();
+	}
+
+	public async getListFolders(commandName: string) {
+		const executionEnvironmentContext = new ExecutionEnvironmentContext({
+			platform: VSCODE_PLATFORM,
+			platformVersion: vscode.version,
+		});
+
+		const listFoldersPromise = AccountFileCabinetService.getFileCabinetFolders(
+			getSdkPath(),
+			executionEnvironmentContext,
+			this.executionPath,
+			commandName
+		);
+		const statusBarMessage = this.translationService.getMessage(LIST_FILES.LOADING_FOLDERS);
+		this.messageService.showStatusBarMessage(statusBarMessage, listFoldersPromise);
+
+		let fileCabinetFolders: FolderItem[] = await listFoldersPromise;
+
+		return this._sortFolders(fileCabinetFolders);
+	}
+
+	private _sortFolders(fileCabinetFolders: FolderItem[]) {
+		fileCabinetFolders = fileCabinetFolders.sort((folder1, folder2) => {
+			if (folder1.isRestricted && !folder2.isRestricted) {
+				return 1;
+			}
+
+			if (!folder1.isRestricted && folder2.isRestricted) {
+				return -1;
+			}
+
+			return 0;
+		});
+		return fileCabinetFolders;
+	}
+
+	public async selectFolder(folders: FolderItem[]): Promise<vscode.QuickPickItem | undefined> {
+		return vscode.window.showQuickPick(
+			folders.map((folder: FolderItem) => {
+				const description = folder.isRestricted ? this.translationService.getMessage(LIST_FILES.RESTRICTED_FOLDER) : '';
+				return { label: folder.path, description };
+			}),
+			{
+				ignoreFocusOut: true,
+				placeHolder: this.translationService.getMessage(LIST_FILES.SELECT_FOLDER),
+				canPickMany: false,
+			}
+		);
+	}
+
+	public async selectFiles(files: string[]): Promise<vscode.QuickPickItem[] | undefined> {
+		return vscode.window.showQuickPick(
+			files.map((file: string) => {
+				const description = file ? this.translationService.getMessage(LIST_FILES.RESTRICTED_FOLDER) : '';
+				return { label: file, description };
+			}),
+			{
+				ignoreFocusOut: true,
+				placeHolder: this.translationService.getMessage(LIST_FILES.SELECT_FOLDER),
+				canPickMany: true,
+			}
+		);
+	}
+
+	public async listFiles(selectedFolder: string, commandName: string) {
+		const listfilesOptions: { [key: string]: string } = {};
+		listfilesOptions[LIST_FILES_COMMAND.OPTIONS.FOLDER] = selectedFolder;
+
+		const commandActionPromise = this.listFilesCommand(listfilesOptions);
+		const commandMessage = this.translationService.getMessage(COMMAND.TRIGGERED, commandName);
+		const statusBarMessage = this.translationService.getMessage(LIST_FILES.LISTING);
+		this.messageService.showInformationMessage(commandMessage, statusBarMessage, commandActionPromise);
+
+		const actionResult = await commandActionPromise;
+		if (actionResult.status === actionResultStatus.SUCCESS) {
+			return actionResult.data;
+		} else {
+			throw actionResult.errorMessages;
+		}
+	}
+
+	private async listFilesCommand(listFilesOption: { [key: string]: string }) {
+		if (!this.vsConsoleLogger) {
+			throw Error('Logger not initialized');
```

Comment:
Should the message be in messages.properties or in a constant?

Reply:
Yes, you have reason. I put this in a constant since it's an internal message that user is not going to see.
Also, I don't think it's needed revision from David Flett.
 What do you think?

Reply:
I agree, no need for David Flett to review it.